sn_node/tests/data_with_churn.rs

```diff
@@ -306,22 +338,33 @@ fn store_chunks_task(client: Client, content: ContentList, churn_period: Duratio
                 .collect();
             let bytes = Bytes::copy_from_slice(&random_bytes);
 
-            let addr = ChunkAddress::new(
-                file_api
-                    .calculate_address(bytes.clone())
-                    .expect("Failed to calculate new Chunk address"),
+            let (addr, chunks) = file_api
+                .chunk_bytes(bytes.clone())
+                .expect("Failed to chunk bytes");
+
+            println!(
+                "Paying storage for ({}) new Chunk/s of file ({} bytes) at {addr:?} in {delay:?}",
+                chunks.len(),
+                bytes.len()
             );
-            println!("Storing Chunk at {addr:?} in {delay:?}");
             sleep(delay).await;
-
-            match file_api
-                .upload_with_proof(bytes, &BTreeMap::default())
+            let proofs = wallet_client
+                .pay_for_storage(chunks.iter().map(|c| c.name()))
                 .await
-            {
+                .expect("Failed to pay for storage for new file at {addr:?}");
+
+            println!(
+                "Storing ({}) Chunk/s of file ({} bytes) at {addr:?} in {delay:?}",
+                chunks.len(),
+                bytes.len()
+            );
+            sleep(delay).await;
+
+            match file_api.upload_with_proof(bytes, &proofs).await {
```

Comment:
Can we call `file_api.upload_chunks_in_batches(chunks.into_iter(), payment_proof, false)` instead to avoid encrypting/converting to chunks again?